android/libraries/rib-base/src/main/java/com/badoo/ribs/core/Router.kt

```diff
@@ -94,7 +94,11 @@ abstract class Router<C : Parcelable, V : RibView>(
     }
 
     fun push(configuration: C) {
-        backStackManager.accept(Push(configuration))
+        if (configuration is Overlay) {
```

Comment:
If it's a flag then any Configuration can be passed both as content-type and overlay-type mode to Router, which I'm not convinced is a good idea. It might make sense to use a RIB differently in different environments, but in the same one? Marking it on the other hand makes this choice unambiguous and explicit.